packages/web-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,194 @@
+   * These values will be applied to css variables that control the tab active indicator position animations
+   */
+  private calculateAnimationProperties(tab: Tab) {
+    this.activeTabOffset = this.getSelectedTabPosition(tab);
+    this.activeTabScale = this.getSelectedTabScale(tab);
+  }
+
+  /**
+   * getSelectedTabPosition - gets the x or y coordinates of the tab
+   */
+  private getSelectedTabPosition(tab: Tab): number {
+    if (this.orientation === 'horizontal') {
+      return this.previousActiveTabData.x - (tab.getBoundingClientRect().x - this.getBoundingClientRect().x);
+    } else return this.previousActiveTabData.y - (tab.getBoundingClientRect().y - this.getBoundingClientRect().y);
+  }
+
+  /**
+   * getSelectedTabScale - gets the scale of the
+   */
+  private getSelectedTabScale(tab: Tab): number {
+    if (this.orientation === 'horizontal') {
+      return this.previousActiveTabData.width / tab.getBoundingClientRect().width;
+    } else return this.previousActiveTabData.height / tab.getBoundingClientRect().height;
+  }
+
+  /**
+   * clearAnimationProperties
+   * resets animation values to defaults. removes the animated css class
+   */
+  private clearAnimationProperties(tab: Tab) {
+    this.previousActiveTabData = { id: '', x: 0, y: 0, height: 0, width: 0 };
+    this.activeTabOffset = 0;
+    this.activeTabScale = 1;
+    tab.classList.remove('animated');
+  }
+
+  /**
+   * animationLoop
+   * runs through all the operations required for setting the tab active indicator to its starting location, ending location, and applying the animated css class to the tab.
+   * @param tab
+   */
+  private animationLoop(tab: Tab) {
+    // If selected, set the animation properties.
+    if (tab.ariaSelected && tab.ariaSelected === 'true') {
+      // get the offset for active indicator
+      // the offset at the start of the animation will be at the location of the previously selected tab
+      this.calculateAnimationProperties(tab);
+      // set the active indicator offset and scale of the active indicator to css vars
+      this.setTabScaleCSSVar();
+      this.setTabOffsetCSSVar();
+
+      // clear properties for next animation
+      this.clearAnimationProperties(tab);
+    }
+
+    this.previousActiveTabData = this.activeTabData;
+    this.calculateAnimationProperties(tab);
+
+    // add the animate css class if the calculated offset is 0 and scale is 1
+    if (this.activeTabOffset === 0 && this.activeTabScale === 1) {
+      tab.classList.add('animated');
+      // update the css vars
+      this.setTabScaleCSSVar();
+      this.setTabOffsetCSSVar();
+      // now when the css class is added the active indicator ::after element will animate
+      // from the previous tab location to its starting location
+    }
+  }
+
+  /**
+   * setTabData
+   * sets the data from the active tab onto the class. used for making all the animation calculations for the active tab indicator.
+   */
+  private setTabData(): void {
```

Comment:
this is a nit, but I'm not sure `set` makes sense here because you're simply calculating things..."set" makes it sound like you're doing work to actually set a value for the component to either provide in the DOM or to a user, neither of which are the case here.

Reply:
The purpose of this method is to save the active tab data onto the class at this.activeTabData. I thought anytime you have a statement where this.classField = someNewValue it is considered "setting."